packages/ia-components/sandbox/audio-players/youtube-player.jsx

```diff
@@ -69,15 +77,81 @@ class YoutubePlayer extends Component {
    * Load video of passed id and default resolution
    */
   componentDidUpdate(prevProps) {
+    const { selectedTrack: capturedTrack } = this.state;
     const { id, selectedTrack } = this.props;
-    const trackChanged = id !== prevProps.id;
-
-    if (trackChanged) {
+    const { id: prevId } = prevProps;
+    const trackChanged = id !== prevId;
+    const playerNotStarted = capturedTrack === null;
+    const setID = !trackChanged && playerNotStarted;
+    if (trackChanged || setID) {
       clearTimeout(this.timer);
       this.setState({ id, selectedTrack }, this.playVideo);
     }
   }
 
+  /**
+   * Remove timer & interval
+   */
+  componentWillUnmount() {
+    clearInterval(this.fullAlbumVideoPoller);
+    clearTimeout(this.timer);
+  }
+
+  /**
+   * Gets available YouTube ID
+   * checks between captured ID (state) & incoming ID (props)
+   * when there isn't a captured ID, it means the player has loaded & user has not selected a video
+   */
+  availableVideoId() {
+    const { id: stateID } = this.state;
+    const { id: propsID } = this.props;
+    const availableID = Number.isInteger(stateID) ? stateID : propsID;
+    return availableID;
+  }
+
+  /**
+   * Check player elapsed time & the track it's on
+   *
+   * @return { object } trackValue
+   * trackValue.elapsedTime = number
+   * trackValue.currentTrack = object (return from fullAlbumDetails)
+   */
+  checkTimeAndTrack() {
+    const { fullAlbumDetails, player } = this.state;
+    const elapsedTime = player.getCurrentTime();
+    const currentTrack = chain(fullAlbumDetails)
+      .filter(track => track.startSeconds < elapsedTime)
+      .last()
+      .value();
+
+    return { elapsedTime, currentTrack };
+  }
+
+  /**
+   * Syncs Full Album Video with Audio Player
+   * - pings YouTube's API to check timestamp
+   * - finds current track position
+   * - sends that to the callback
+   */
+  syncVideoWithPlayer() {
+    const { videoStartedPlaying } = this.state;
+    const { youtubePlaylistChange } = this.props;
+    const setURLOnly = true;
+
+    const videoTimePoller = () => {
+      const { currentTrack } = this.checkTimeAndTrack();
+      const { trackNumber } = currentTrack;
+      youtubePlaylistChange(trackNumber, setURLOnly);
+    };
+
+    if (videoStartedPlaying) {
+      videoTimePoller();
+      if (!this.fullAlbumVideoPoller) {
+        this.fullAlbumVideoPoller = setInterval(videoTimePoller, 5000);
```

Comment:
QUESTION/SUGGESTION: What is the significance of this time value, 5000ms? i.e. Why this value? Maybe add a comment.

Reply:
A few notes: on full screen, there is a lil message that displays at the lower left corner of the video every time a request happens. Didn't look too good when it happened to flash every second. With our selected videos going to for minutes at a time, 5 seconds "felt" about the right amount of time to keep the track listing in sync with track changes.  will add a comment.